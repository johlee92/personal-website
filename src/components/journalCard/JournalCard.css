.journalCard {
    width: 100%;
    border: 2px darkgray solid;
    max-width: 720px;
    margin-bottom: 25px;
}

.journalCard h2{
    margin: 0;
}

.journalCard-container {
    padding: 10px;
}

button.journalCard-button {
    width: 120px;
    height: 30px;
    font-family: 'Noto Sans JP', sans-serif;
    text-align: center;
    font-size: 16px;
    cursor: pointer;
    border: 3px black solid;
    background-color: white;
    margin-right: 10px;
    margin-top: 5px;
}

button.journalCard-button:hover {
    color: white;
    background-color: darkgray;
}