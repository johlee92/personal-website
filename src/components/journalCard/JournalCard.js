import React, { Component } from 'react';
import './JournalCard.css';
import { Link } from 'react-router-dom';
import moment from 'moment';

class JournalCard extends Component {
    render() {
        // creates the path for the journal that is selected
        const journalPath = `/journals/${this.props.id}`;

        return (
            <div className="journalCard">
                <div className="journalCard-container">
                    <h2>{this.props.name}</h2>
                    <p>{moment(this.props.created).format('LL')}</p>
                    <p>{this.props.short}</p>
                    <Link to={journalPath}><button type="button" className="journalCard-button">Read More</button></Link>
                </div>
            </div>
        )
    }
}

export default JournalCard;