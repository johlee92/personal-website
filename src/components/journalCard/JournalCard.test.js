import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import JournalCard from './JournalCard';

describe(`JournalCard component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
                <JournalCard />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <JournalCard />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
