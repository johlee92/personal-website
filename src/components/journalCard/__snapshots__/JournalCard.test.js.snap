// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalCard component renders the component as expected 1`] = `
<div
  className="journalCard"
>
  <div
    className="journalCard-container"
  >
    <h2>
      Dogs
    </h2>
    <p>
      January 2, 2019
    </p>
    <p>
      short description of the journal
    </p>
    <a
      href="/journals/1"
      onClick={[Function]}
    >
      <button
        className="journalCard-button"
        type="button"
      >
        Read More
      </button>
    </a>
  </div>
</div>
`;
